mne/io/reference.py

```diff
@@ -97,11 +128,16 @@ def _apply_reference(inst, ref_from, ref_to=None):
                 'using the apply_proj() method function before the desired '
                 'reference can be set.'
             )
+
     for i in projs_to_remove:
         del inst.info['projs'][i]
 
-    ref_from = [inst.ch_names.index(ch) for ch in ref_from]
-    ref_to = [inst.ch_names.index(ch) for ch in ref_to]
+    # Need to call setup_proj after changing the projs:
+    inst._projector, _ = \
+        setup_proj(inst.info, add_eeg_ref=False, activate=False)
```

Comment:
this is the line that was missing. Since the inst.info['projs'] is hacked just above then the `_projector` attribute was not up to date.